src/ert/config/queue_config.py

```diff
@@ -139,17 +113,130 @@ def generate_dict(option_list: List[Tuple[str, str]]) -> Dict[str, List[str]]:
+        "EXCLUDE_HOST",
+    ],
+    "TORQUE": [
+        "QSUB_CMD",
+        "QSTAT_CMD",
+        "QDEL_CMD",
+        "QSTAT_OPTIONS",
+        "QUEUE",
+        "DEBUG_OUTPUT",
+    ],
+    "LOCAL": [],
+}
+
+positive_int_options: Mapping[str, List[str]] = {
+    "LSF": [
+        "BJOBS_TIMEOUT",
+        "MAX_RUNNING",
+    ],
+    "SLURM": [
+        "MEMORY",
+        "MEMORY_PER_CPU",
+        "MAX_RUNNING",
+    ],
+    "TORQUE": [
+        "NUM_NODES",
+        "NUM_CPUS_PER_NODE",
+        "MAX_RUNNING",
+    ],
+    "LOCAL": ["MAX_RUNNING"],
+}
+
+float_options: Mapping[str, List[str]] = {
+    "LSF": [
+        "SUBMIT_SLEEP",
+    ],
+    "SLURM": [
+        "SQUEUE_TIMEOUT",
+    ],
+    "TORQUE": ["SUBMIT_SLEEP", "QUEUE_QUERY_TIMEOUT"],
+    "LOCAL": [],
+}
+
+bool_options: Mapping[str, List[str]] = {
+    "LSF": ["DEBUG_OUTPUT"],
+    "SLURM": [],
+    "TORQUE": [],
+    "LOCAL": [],
+}
+
+
+def _validate_queue_options(
+    queue_options: List[Tuple[str, str]], queue_type: str
+) -> None:
+    for option_strings in queue_options:
         option_name = option_strings[0]
         option_value = option_strings[1]
         if (
             option_value != ""  # This is equivalent to the option not being set
-            and option_name == "MEMORY_PER_JOB"
+            and option_name in memory_options[queue_type]
             and re.match("[0-9]+[mg]b", option_value) is None
         ):
             raise ConfigValidationError(
-                f"The value '{option_value}' is not valid for the Torque option "
+                f"The value '{option_value}' is not valid for "
+                f"the {queue_type} option "
                 "MEMORY_PER_JOB, it must be of "
                 "the format '<integer>mb' or '<integer>gb'."
             )
+        if option_name in string_options[queue_type] and not isinstance(
+            option_value, str
+        ):
+            raise ConfigValidationError(
+                f"The value '{option_value}' is not valid string "
+                f"for the {queue_type} option "
+                f"{option_name}, it must be of string type."
+            )
+        if option_name in float_options[queue_type] and (
+            re.match(r"^\d+(\.\d+)?$", option_value) is None and option_value != ""
+        ):
+            raise ConfigValidationError(
+                f"The value '{option_value}' is not valid integer or float"
+                f" for the {queue_type} option {option_name}."
+            )
+        if option_name in positive_int_options[queue_type] and (
+            re.match(r"^\d+$", option_value) is None and option_value != ""
+        ):
+            raise ConfigValidationError(
+                f"The value '{option_value}' is not valid positive integer "
+                f" for the {queue_type} option {option_name}."
+            )
+        if option_name in bool_options[queue_type] and not option_value in [
+            "TRUE",
+            "FALSE",
+            "",
```

Comment:
My gut tells me that we should perhaps allow lower-case values of this. Maybe also `0` and `1`. Yes/no?